src/main/java/com/cloudbees/jenkins/plugins/sshagent/RemoteAgent.java

```diff
@@ -45,10 +45,10 @@
      * @param comment    the comment to give to the key.
      * @throws java.io.IOException if something went wrong.
      */
-    void addIdentity(String privateKey, String passphrase, String comment) throws IOException;
+    void addIdentity(String privateKey, String passphrase, String comment) throws IOException, InterruptedException;
 
     /**
      * Stops the agent.
      */
-    void stop();
+    void stop() throws IOException, InterruptedException;
```

Comment:
These methods may now be running remote commands, so they should be allowed to throw reasonable exceptions.